### PR TITLE
prior-guided acquisition function

### PR DESCRIPTION
**Pull request by sdaulton**
Summary: This adds an acquisition function wrapper for prior-guided AFs.

Differential Revision: D47248296

### CODE REVIEW

File: botorch/acquisition/fixed_feature.py [hunk 1]
```diff
@@ -20,7 +22,31 @@
 from torch.nn import Module
 
 
-class FixedFeatureAcquisitionFunction(AcquisitionFunction):
+class AcquistionFunctionWrapperMixin(ABC):
```
Comment (Balandat) [inline botorch/acquisition/fixed_feature.py:25]: This is a very generic name for something rather specific, can name this differently? Like `PendingPointAcqfWrapperMixin` (or some better name)?

File: botorch/acquisition/fixed_feature.py [hunk 2]
```diff
@@ -20,7 +22,31 @@
...
+            raise ValueError(
+                f"Base acquisition function {type(self.acq_func).__name__} "
+                "does not have an `X_pending` attribute."
+            )
+
+    @X_pending.setter
+    def X_pending(self, X_pending: Optional[Tensor]):
+        r"""Sets the `X_pending` of the base acquisition function."""
+        if X_pending is not None:
+            self.acq_func.X_pending = self._construct_X_full(X_pending)
```
Comment (Balandat) [inline botorch/acquisition/fixed_feature.py:41]: seems like `_construct_X_full` should be an abstract method on this mixin?

File: botorch/acquisition/prior_guided.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+from __future__ import annotations
+
+from botorch.acquisition.acquisition import AcquisitionFunction
+from botorch.acquisition.fixed_feature import AcquistionFunctionWrapperMixin
+from torch import Tensor
+
+from torch.nn import Module
+
+
+class PriorGuidedAcquisitionFunction(
```
Comment (Balandat) [inline botorch/acquisition/prior_guided.py:27]: Need to implement `_construct_X_full` here?

File: botorch/acquisition/fixed_feature.py (hunk 1 above)
Comment (sdaulton) [reply botorch/acquisition/fixed_feature.py:25]: Ah I'll remove this. The set_X_pending functionality was not as generic as I thought

**Comment by sdaulton** [comment]
FYI @hvarfner

**Comment by eytan** [comment]
cc @hvarfner.

**Pull request closed without merging**